drafts/inlinemacros.md

```diff
@@ -62,16 +65,19 @@ is equivalent to
 
 ## Inline Reductions
 
-The compiler will do the following rewritings.
+The compiler will do the following rewritings when encountering certain patterns of code
+outside the bodies of inline values and inline methods.
+When these patterns of code appear inside the aforementioned constructs, 
+they will be ignored by the inliner and processed by the compiler as usual.
```

Comment:
Do we expand inline applications inside other inline methods? 

```
inline def foo() = ...
inline def bar() = foo() // (1)
bar // (2)
// when do we evaluate ...? at (1) or at (2)?
```

What about inline values?

```
inline val bar = ... // (1)
inline def foo() = bar // (2)
foo() // (3)
// when do we evaluate ...? at (1) or at (2) or at (3)?
```

Do we expand inline methods and values inside macro expressions? 

```
inline def foo() = ...
inline def bar() = macro(foo()) // (1)
bar() // (2)
// when do we evaluate ...? at (1) or at (2)
```

These are the easier questions. There are also combinations. What if we have a standalone macro expression whose body refers to an inline val whose body calls an inline method? When do these things expand?